sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4236,9 +4350,14 @@ pi_result piEventRelease(pi_event Event) {
     if (Event->CommandType == PI_COMMAND_TYPE_MEM_BUFFER_UNMAP &&
         Event->CommandData) {
       // Free the memory allocated in the piEnqueueMemBufferMap.
-      ZE_CALL(zeMemFree,
-              (Event->Queue->Context->ZeContext, Event->CommandData));
-      Event->CommandData = nullptr;
+      if (IndirectAccessTrackingEnabled) {
+        // Use the version with reference counting
+        PI_CALL(piextUSMFree(Event->Queue->Context, Event->CommandData));
```

Comment:
I don't quite understand this change. Can we always do the `piextUSMFree`?

Reply:
Also, why do you not clear the `Event->CommandData` on this path?

Reply:
Yes, we can. I've put in under guard because all changes of this PR are under the guard.

Reply:
It is a mistake with Event->CommandData, let me fix it.